clusterbuster

```diff
@@ -2351,6 +2351,86 @@ EOF
     done
 }
 
+function create_vm_deployment() {
+    local namespace=$1
+    local instance=$2
+    local -i replicas=$4
+    local -i replica=0
+    local depname="${namespace}-${workload}-${instance}"
+    while (( replica++ < replicas )) ; do
+        local name="${depname}-${replica}"
+        local vmname=$(mkpodname "$name")
+        create_object -n "$namespace" "$name" <<EOF
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  labels:
+    app: $name
+    kubevirt-vm: $vmname
+  name: $vmname
+$(indent 2 standard_deployment_metadata_yaml "$namespace" client)
+$(indent 2 standard_pod_metadata_yaml "$namespace" client)
+$(create_vm_spec -n "$vmname")
+EOF
+    done
+}
+
+function create_vm_spec() {
+    local vmname cores sockets threads mem grace image password
+    local OPT OPTIND=1 OPTARG
+    while getopts ':n:c:s:t:m:g:i:p:-' OPT; do
+        case "$OPT" in 
+        ('?') fatal "Unknown option -$OPTARG" ;;
+        ( : ) fatal "Option -$OPTARG requires a value" ;;
+        ( - ) break ;; # end options with --
+        ( n ) vmname=$OPTARG ;;
+        ( c ) cores=$OPTARG ;;
+        ( s ) sockets=$OPTARG ;;
+        ( t ) threads=$OPTARG ;;
+        ( m ) mem=$OPTARG ;;
+        ( g ) grace=$OPTARG ;;
+        ( i ) image=$OPTARG ;;
+        ( p ) password=$OPTARG ;;
+        ( * ) fatal "Unhandled option -$OPT"
+        esac
+    done
+    shift $((OPTIND-1))
+    [[ -z $vmname ]] && fatal "vmname is required"
+    cores=${cores:-1}; sockets=${sockets:-1}; threads=${threads:-1}
+    mem=${mem:-1024}; grace=${grace:-30}; password=${password:-fedora}
+    image=${image:-quay.io/kubevirt/fedora-container-disk-images:35}
+    cat <<EOF
+spec:
+  running: true
+  template:
+    metadata:
+      labels:
+        kubevirt-vm: $vmname
+    spec:
+      domain:
+        cpu:
+          cores: $cores
+          sockets: $sockets
+          threads: $threads
+        devices:
+          disks:
+            - name: containerdisk
+              disk:
+                bus: virtio   
+        machine:
+          type: q35
```

Comment:
What does "q35" mean here?

Reply:
It is the newest x86_64 chipset that QEMU supports, used in KubeVirt by default. I think I can also remove this line.